syntax = "proto3";

package frauddetection;

service FraudDetectionService {
    rpc CheckFraud (FraudRequest) returns (FraudResponse);
    rpc HealthCheck (HealthCheckRequest) returns (HealthCheckResponse);
}

message FraudRequest {
    string name = 1;
    string email = 2;
    string address = 3;
    string phone = 4;
    string ssn = 5;
}

message FraudResponse {
    bool isFraud = 1;
    string message = 2;
}

message HealthCheckRequest {
}

message HealthCheckResponse {
    string status = 1;
}